### PR TITLE
Add Confluent Schema Registry to requirements section

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Change to documentation only.

> How would you describe this change to a non-technical end user or system administrator?

Add Confluent Schema Registry to requirements section.

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/kafka.rst [hunk 1]
```diff
@@ -380,14 +380,16 @@ There is no limit on field descriptions for either key or message.
...
-
-The benefits of using the schema registry is that new tables can be defined
-without a cluster restart, schema updates are detected and there is no need to
-define tables manually.
+The Kafka connector only uses the `Confluent Schema Registry
+<https://docs.confluent.io/1.0/schema-registry/docs/intro.html>`_ to
+discover table definitions. The benefits of using the schema registry are:
+
+* New tables can be defined without a cluster restart.
+* Schema updates are detected.
```
Comment (Jessie212) [inline docs/src/main/sphinx/connector/kafka.rst:388]: @hashhar ... are detected how? Automatically?

Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:388]: Automatically. It's a black from our perspective and user's too. The Kafka library Confluent provides handles all of this.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -380,14 +380,16 @@ There is no limit on field descriptions for either key or message.
...
-<https://docs.confluent.io/1.0/schema-registry/docs/intro.html>`_ to discover
-table definitions. Set ``kafka.table-description-supplier`` to ``CONFLUENT`` to
-use it and configure the additional properties in the following table.
-
-The benefits of using the schema registry is that new tables can be defined
-without a cluster restart, schema updates are detected and there is no need to
-define tables manually.
+The Kafka connector only uses the `Confluent Schema Registry
+<https://docs.confluent.io/1.0/schema-registry/docs/intro.html>`_ to
+discover table definitions. The benefits of using the schema registry are:
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/kafka.rst:385]: If the intent is to indicate it only works with the Confluent Schema Registry then would this be better? Current wording seems confusing and doesn't convey what we want to convey.

```suggestion
The Confluent table description supplier uses the `Confluent Schema Registry
<https://docs.confluent.io/1.0/schema-registry/docs/intro.html>`_ to discover
table definitions. Set ``kafka.table-description-supplier`` to ``CONFLUENT`` to
use it and configure the additional properties in the following table.
This table description supplier is only tested to work with the Confluent Schema Registry [but may or may not work with other compatible schema registries].
```

Feel free to drop the part in `[]` depending on what we want to convey.
```

File: docs/src/main/sphinx/connector/kafka.rst [hunk 3]
```diff
@@ -382,12 +382,17 @@ Confluent table description supplier
 
 The Confluent table description supplier uses the `Confluent Schema Registry
 <https://docs.confluent.io/1.0/schema-registry/docs/intro.html>`_ to discover
-table definitions. Set ``kafka.table-description-supplier`` to ``CONFLUENT`` to
-use it and configure the additional properties in the following table.
+table definitions. It is only tested to work with the Confluent Schema
+Registry.
```
Comment (Jessie212) [inline docs/src/main/sphinx/connector/kafka.rst:386]: @hashhar I've incorporated part of your suggestion. Wdyt?

File: docs/src/main/sphinx/connector/kafka.rst [hunk 4]
```diff
@@ -382,12 +382,17 @@ Confluent table description supplier
...
 
-The benefits of using the schema registry is that new tables can be defined
-without a cluster restart, schema updates are detected and there is no need to
-define tables manually.
+The benefits of using the schema registry are:
+
+* New tables can be defined without a cluster restart.
+* Schema updates are detected automatically.
+* There is no need to define tables manually.
```
Comment (Jessie212) [inline docs/src/main/sphinx/connector/kafka.rst:392]: Is it necessary to list the benefits of the Confluent Schema Registry if the user doesn't have a choice, in this particular instance?

Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:392]: It's not the benefits of the "Confluent" schema registry itself. It's instead contrasting this to the `File table description supplier`. I agree it's unclear since it's a long document and the heirarchy seems to be lost.

Maybe reword to "The benefits of using the Confluent table description supplier instead of file table description supplier...".

Comment (Jessie212) [reply docs/src/main/sphinx/connector/kafka.rst:392]: Okay, so using the Confluent Schema Registry over the default?

Comment (hashhar) [reply docs/src/main/sphinx/connector/kafka.rst:393]: Yes.

**Review by hashhar — APPROVED** [review]

**Review by mosabua — APPROVED** [review]
